cpp/src/gandiva/precompiled/time.cc

```diff
@@ -689,21 +691,47 @@ timestamp castTIMESTAMP_utf8(int64_t context, const char* input, int32 length) {
 
 timestamp castTIMESTAMP_date64(date64 date_in_millis) { return date_in_millis; }
 
-char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 length,
-                                  int32* out_len) {
-  std::string timestamp_str = std::to_string(in);
+const char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 length,
+                                        int32* out_len) {
+  int64 year = extractYear_timestamp(in);
+  int64 month = extractMonth_timestamp(in);
+  int64 day = extractDay_timestamp(in);
+  int64 hour = extractHour_timestamp(in);
+  int64 minute = extractMinute_timestamp(in);
+  int64 second = extractSecond_timestamp(in);
+  int64 millis = in % MILLIS_IN_SEC;
+
+  const int full_output_length = 23;
```

Comment:
please make this a static const. per the [convention](https://google.github.io/styleguide/cppguide.html#Constant_Names), the name should be like  kTimeStampStringLen

Reply:
done. Thanks